data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/strategy/UnorderedConsumerRecordOffsetStrategy.java

```diff
@@ -57,64 +66,98 @@ public UnorderedConsumerRecordOffsetStrategy(final KafkaConsumer<?, ?> consumer,
   @Override
   public void recordReceived(final KafkaConsumerRecord<?, ?> record) {
     // un-ordered processing doesn't require pause/resume lifecycle.
+
+    // Because recordReceived is guaranteed to be called in order,
+    // we use it to set the last seen acked offset.
+    // TODO If this assumption doesn't work, use this.consumer.committed(new TopicPartition(record.topic(), record.partition()))
+    this.lastAckedMap.putIfAbsent(new TopicPartition(record.topic(), record.partition()), record.offset() - 1);
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
   public void successfullySentToSubscriber(final KafkaConsumerRecord<?, ?> record) {
-    // TODO evaluate if it's worth committing offsets at specified intervals per partition.
-    // commit each record
-    commit(record)
-      .onSuccess(ignored -> {
-        eventsSentCounter.increment();
-        logger.debug(
-          "committed {} {} {}",
-          keyValue("topic", record.topic()),
-          keyValue("partition", record.partition()),
-          keyValue("offset", record.offset())
-        );
-      })
-      .onFailure(cause -> logger.error(
-        "failed to commit {} {} {}",
-        keyValue("topic", record.topic()),
-        keyValue("partition", record.partition()),
-        keyValue("offset", record.offset()),
-        cause
-      ));
+    commit(record);
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
   public void successfullySentToDLQ(final KafkaConsumerRecord<?, ?> record) {
-    successfullySentToSubscriber(record);
+    commit(record);
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
   public void failedToSendToDLQ(final KafkaConsumerRecord<?, ?> record, final Throwable ex) {
-    // do not commit
+    // TODO what do we do with that?
```

Comment:
IMO here we should commit, or at least we should modify the internal state of this class in a way such that on the next "successful message", it will be committed (aka adding this commit to the `pendingAcksMap`). Otherwise, on the topic partition of this record, no commit will ever be committed again.

Reply:
I agree we should update the internal state.